code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm

```diff
@@ -0,0 +1,1553 @@
+
+/// Stops processing, and unregisters related signals.
+/datum/action/xeno_action/primal_wrath/proc/stop_ability(datum/source)
+	SIGNAL_HANDLER
+	if(ability_active)
+		toggle_buff(FALSE)
+	UnregisterSignal(owner, list(COMSIG_PARENT_QDELETING, COMSIG_MOB_DEATH, COMSIG_XENOMORPH_EVOLVED, COMSIG_XENOMORPH_DEEVOLVED))
+	STOP_PROCESSING(SSprocessing, src)
+
+
+// ***************************************
+// *********** Earth Pillar (also see: Earth Riser)
+// ***************************************
+#define EARTH_PILLAR_KNOCKDOWN_DURATION 0.5 SECONDS
+#define EARTH_PILLAR_DAMAGE_MECHA_MODIFIER 5
+#define EARTH_PILLAR_DAMAGE_OBJECT_MODIFIER 2
+#define EARTH_PILLAR_DAMAGE_TURF_MODIFIER 5
+
+/particles/earth_pillar
+	icon = 'icons/effects/particles/generic_particles.dmi'
+	icon_state = "behemoth_smoke"
+	width = 100
+	height = 100
+	count = 1000
+	spawning = 3
+	gravity = list(0, 0.15)
+	lifespan = 15
+	fade = 15
+	position = generator(GEN_SPHERE, 0, 17, UNIFORM_RAND)
+	velocity = list(0, 0.2)
+	scale = generator(GEN_NUM, 0.3, 0.5, UNIFORM_RAND)
+	grow = 0.05
+	spin = generator(GEN_NUM, 10, 20)
+
+/obj/effect/temp_visual/behemoth/earth_pillar
+	duration = 2 SECONDS
+	layer = ABOVE_ALL_MOB_LAYER
+
+/obj/effect/temp_visual/behemoth/earth_pillar/projectile
+	icon = 'icons/effects/128x128.dmi'
+	icon_state = "earth_pillar_exploded"
+	duration = 1.6 SECONDS
+	layer = ABOVE_ALL_MOB_LAYER
+	pixel_x = -48
+	pixel_y = -41
+
+/obj/effect/temp_visual/behemoth/earth_pillar/projectile/Initialize(mapload)
+	. = ..()
+	animate(src, time = duration - 0.5 SECONDS)
+	animate(alpha = 0, time = 0.5 SECONDS, easing = CIRCULAR_EASING|EASE_OUT)
+
+/obj/effect/temp_visual/behemoth/earth_pillar/destroyed
+	icon = 'icons/effects/128x128.dmi'
+	icon_state = "earth_pillar_destroyed"
+	duration = 1 SECONDS
+	pixel_x = -49
+	pixel_y = -38
+
+/obj/effect/temp_visual/behemoth/earth_pillar/destroyed/Initialize(mapload)
+	. = ..()
+	animate(src, time = 0.3 SECONDS)
+	animate(alpha = 0, time = 0.7 SECONDS, easing = CIRCULAR_EASING|EASE_OUT)
+
+/obj/effect/temp_visual/behemoth/earth_pillar/broken
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "earth_pillar_broken"
+	duration = 0.9 SECONDS
+	pixel_x = -32
+	pixel_y = -29
+
+/obj/effect/temp_visual/behemoth/earth_pillar/broken/Initialize(mapload)
+	. = ..()
+	animate(src, alpha = 0, time = duration, easing = CIRCULAR_EASING|EASE_OUT)
+
+/obj/structure/earth_pillar
+	name = "earth pillar"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "earth_pillar"
+	base_icon_state = "earth_pillar"
+	layer = ABOVE_LYING_MOB_LAYER
+	climbable = TRUE
+	climb_delay = 3 SECONDS
+	interaction_flags = INTERACT_CHECK_INCAPACITATED
+	density = TRUE
+	max_integrity = 300
+	soft_armor = list(MELEE = 50, BULLET = 50, LASER = 50, ENERGY = 50, BOMB = 0, BIO = 100, FIRE = 100, ACID = 0)
+	destroy_sound = 'sound/effects/behemoth/earth_pillar_destroyed.ogg'
+	coverage = 128
+	/// The owner of this object.
+	var/owner
```

Comment:
If the behemoth gets deleted all these pillars will not be deleted so that owner = null and will cause a hard delete

You can fix this by QDEL_LIST the pillars list + add owner = null to pillar Destroy when you lose the ability I guess?

Reply:
Sure, donezo.